mingw_files/nrnmingwenv.sh

```diff
@@ -185,4 +184,5 @@ if test -f /mingw64/lib/libgcc_s.a ; then # gcc 11.2.0 Rev 10
 fi
 copy $gcclib '
 libgcc_s.a
+libstdc++.dll.a
```

Comment:
Thanks. I'm updating my msys which had ```/mingw64/lib/gcc/x86_64-w64-mingw32/11.2.0/libstdc++.dll.a``` and also in a few other places. After the update there is only ```/mingw64/lib/libstdc++.dll.a```

Reply:
@nrnhines is there anything else pending on the windows side?

Reply:
ah, since we have disabled the RxD testing, I guess we should manually trigger those
https://github.com/neuronsimulator/nrn/blob/master/ci/win_test_installer.cmd#L55

Reply:
I'm experiencing a problem on my windows guest  with the last step of pre_setup_exe.sh with ```cd $N/demo/release ; $N/lib/mknrndl2.sh``` where, although N is exported, it does not make it into  mknrndl2.sh. When that is resolved, I will turn my attention to an issue I'm having with nrn-modeldb-ci involving subprocesses. I.e.
```
run_commands(model, ["/bin/sh", "-c", "rm -rf ./x86_64/"], work_dir=model.run_info["start_dir"])
```
is failing. It may be this weekend before I resolve everything.

Reply:
> ah, since we have disabled the RxD testing, I guess we should manually trigger those https://github.com/neuronsimulator/nrn/blob/master/ci/win_test_installer.cmd#L55

For RxD looks like we should still keep the CC toolchain. 
@ramcdougal is it possible to migrate the reaction compilation to CPP? 
https://github.com/neuronsimulator/nrn/blob/master/share/lib/python/neuron/rxd/rxd.py#L522

```
cabuf:
Traceback (most recent call last):
  File "c:\nrnCPP\lib\python\neuron\rxd\rxd.py", line 529, in _c_compile
    gcc = os.environ["CC"]
  File "C:\Python39\lib\os.py", line 679, in __getitem__
    raise KeyError(key) from None
KeyError: 'CC'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\nrnCPP\lib\python\neuron\rxd\rxd.py", line 1849, in _init
    initializer._do_init()
  File "c:\nrnCPP\lib\python\neuron\rxd\initializer.py", line 58, in _do_init
    rxd._init()
  File "c:\nrnCPP\lib\python\neuron\rxd\rxd.py", line 1872, in _init
    _compile_reactions()
  File "c:\nrnCPP\lib\python\neuron\rxd\rxd.py", line 1484, in _compile_reactions
    _c_compile(fxn_string),
  File "c:\nrnCPP\lib\python\neuron\rxd\rxd.py", line 539, in _c_compile
    raise RxDException(
neuron.rxd.rxdException.RxDException: unable to locate a C compiler. Please `set CC=<path to C compiler>`

NEURON: Python Callback failed
```

Reply:
I'll experiment with the cython ```--cplus``` arg.

My problem with the missing bash environment when pre_setup_exe.sh calls mknrndl2.sh is called is being strangely resistant to solving.  I concluded the basic problem was that changing the PATH so that the neurondemo build could test the copied c++ tool chain is using a different bash to run mknrndl2.sh than the bash that is running the pre_setup_exe.sh. This results in the environment for mknrndl2.sh being almost empty. Don't know why this problem would appear on the latest msys update when its' been working for more than a decade. My current work around almost works. Ie. calling a wrapper script with args for N, PATH, MODLUNIT, TMP, and TEMP, so that the wrapper can export those and call mknrndl2.sh. That works for an explicit test of
``` 
bash /e/hines/neuron/nmodlcxx/build/src/mswin/pre_setup_exe.sh
```
But with
```
ninja setup_exe
```
there is a bizarre concatenation of $N and $TMP that causes failure. (note that for testing, mknrndl2a.sh prints the environment just before calling mknrndl2.sh)
```
+ E:/hines/neuron/nmodlcxx/src/mswin/lib/mknrndl2a.sh C:/marshalnrn/nrn /proc/cygdrive/c/marshalnrn/nrn/bin:/proc/cygdrive/c/marshalnrn/nrn/mingw/usr/bin:/proc/cygdrive/c/marshalnrn/nrn/mingw/mingw64/bin:/e/hines/bin:/mingw64/bin:/usr/local/bin:/usr/bin:/usr/bin:/c/Windows/System32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl C:/marshalnrn/nrn/lib/nrnunits.lib /c/Users/hines/AppData/Local/Temp /c/Users/hines/AppData/Local/Temp
MSYSTEM=MINGW64
PWD=/cygdrive/c/marshalnrn/nrn/demo/release
N=C:/marshalnrn/nrn
HOME=/home/hines
SYSTEMROOT=C:\WINDOWS
TERM=xterm-256color
TEMP=/c/Users/hines/AppData/Local/Temp
SHLVL=1
MODLUNIT=C:/marshalnrn/nrn/lib/nrnunits.lib
TMP=/c/Users/hines/AppData/Local/Temp
PATH=/proc/cygdrive/c/marshalnrn/nrn/bin:/proc/cygdrive/c/marshalnrn/nrn/mingw/usr/bin:/proc/cygdrive/c/marshalnrn/nrn/mingw/mingw64/bin:/e/hines/bin:/mingw64/bin:/usr/local/bin:/usr/bin:/usr/bin:/c/Windows/System32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
WINDIR=C:\WINDOWS
SYSTEMDRIVE=C:
_=/proc/cygdrive/c/marshalnrn/nrn/mingw/usr/bin/env
x86_64-w64-mingw32-g++.exe -DDLL_EXPORT -DPIC -IC:/marshalnrn/nrn/include -IC:/marshalnrn/nrn/src/scopmath -IC:/marshalnrn/nrn/src/nrnoc -IC:/marshalnrn/nrn/src/oc  -c mod_func.cpp
Cannot create temporary file in C:\marshalnrn\nrn\mingw\c\Users\hines\AppData\Local\Temp\: No such file or directory
make: *** [C:/marshalnrn/nrn/lib/mknrndll.mak:65: mod_func.o] Error 3
ninja: build stopped: subcommand failed.
```

Reply:
It appears that a solution to the TMP problem is to have the wrapper set TMP explicitly. ie.
```
export TMP="$N/mingw/tmp"
```

Reply:
As far as compilation goes, should just work as long as we:
- have a way to find a C++ compiler
- `extern "C"` things so that `ctypes` can load them

Reply:
Ok so I've quickly tried to use the C++ compiler but it wasn't straightforward. I've put back the CC toolchain and re-launched RxD locally (all work OK). We can open a follow-up ticket for the CPP migration